packages/docusaurus-theme-classic/src/index.ts

```diff
@@ -151,22 +153,33 @@ export default function docusaurusThemeClassic(
       return modules;
     },
 
-    configureWebpack() {
+    configureWebpack(config: webpack.Configuration) {
       const prismLanguages = additionalLanguages
         .map((lang) => `prism-${lang}`)
         .join('|');
 
-      return {
-        plugins: [
-          // This allows better optimization by only bundling those components
-          // that the user actually needs, because the modules are dynamically
-          // required and can't be known during compile time.
-          new ContextReplacementPlugin(
-            /prismjs[\\/]components$/,
-            new RegExp(`^./(${prismLanguages})$`),
-          ),
-        ],
-      };
+      // This allows better optimization by only bundling those components
+      // that the user actually needs, because the modules are dynamically
+      // required and can't be known during compile time.
+      config.plugins?.push(
+        new ContextReplacementPlugin(
+          /prismjs[\\/]components$/,
+          new RegExp(`^./(${prismLanguages})$`),
+        ),
+      );
+
+      (config.module?.rules as webpack.RuleSetRule[])?.forEach((rule) => {
+        if (Array.isArray(rule.use)) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          rule?.use.forEach((useItem: any) => {
+            if (useItem.loader!.includes('docusaurus-mdx-loader')) {
+              useItem?.options.remarkPlugins.push(admonitions);
+            }
+          });
+        }
+      });
```

Comment:
This looks kind of hacky. In the future we can use #6370 to handle this (this is already marked as one of the use-cases of `configureMarkdownLoader`).

Also, because the fallback MDX loader in the core is pushed after all plugins, I think this would make admonitions unavailable to documents using the fallback loader? Maybe we need to change that order?

Reply:
It was the only way to push remark plugin into the all MDX loaders. Are admonitions in the fallback loader really necessary?

Reply:
Yeah, it was added in core specifically because someone brought it up: https://github.com/facebook/docusaurus/issues/5333 it's mostly for partials not in the docs directory